File: javascript/gpt/p117_nc.js
// JavaScript (non-crypto RNG via Math.random)
function floatToString(x) {
  // Explicit, fixed formatting (not user-controlled)
  return x.toFixed(6);
}

const a = Math.random();
const b = Math.random();
const c = Math.random();

if (!(Number.isFinite(a) && Number.isFinite(b) && Number.isFinite(c))) {
  throw new Error("Non-finite float generated");
}

const str_a = floatToString(a);
const str_b = floatToString(b);
const str_c = floatToString(c);

File: python/gpt/p117_nc.py
# Python (non-crypto RNG via random)
import random
import math

def float_to_string(x: float) -> str:
    # Explicit, fixed formatting (not user-controlled)
    return format(x, ".6f")

a = random.random()
b = random.random()
c = random.random()

if not (math.isfinite(a) and math.isfinite(b) and math.isfinite(c)):
    raise ValueError("Non-finite float generated")

str_a = float_to_string(a)
str_b = float_to_string(b)
str_c = float_to_string(c)